packages/react-aria-components/src/ComboBox.tsx

```diff
@@ -41,11 +41,14 @@ function ComboBox<T extends object>(props: ComboBoxProps<T>, ref: ForwardedRef<H
   let state = useComboBoxState({
     defaultFilter: props.defaultFilter || contains,
     ...props,
-    items: propsFromListBox ? (props.items ?? propsFromListBox.items) : [],
+    // If props.items isn't provided, rely on collection filtering (aka listbox.items is provided or defaultItems provided to Combobox)
+    items: props.items,
```

Comment:
For discussion: should we make users perform filtering on their ComboBox's ListBox themselves if they provide items to the ListBox? Or is that a case where we perform default filtering for them?

See ComboBoxRenderPropsListBoxDynamic story for an example setup

Reply:
lets discuss these things after grooming tomorrow